### ISSUE TITLE
TPCDS extension generates string columns that do not match the standard

### ISSUE THREAD

**Issue by DanCory** [duckdb/duckdb#8584]
### What happens?

Use the TPCDS extension to generate data. It doesn't match the data provided by the current version of the TPCDS standard.

The extension is based on version 2.10 of the standard's generation code, while the current version is 3.2. Unfortunately I do not have an old version of the TPCDS generation code to compare to.

But it could also have to do with the porting of the extension to C++.

### To Reproduce

Run the following SQL:
.open sf1
call dsdgen(sf=1);
-- TPCDS query 37
select  i_item_id
       ,i_item_desc
       ,i_current_price
 from item, inventory, date_dim, catalog_sales
 where i_current_price between 68 and 68 + 30
 and inv_item_sk = i_item_sk
 and d_date_sk=inv_date_sk
 and d_date between cast('2000-02-01' as date) and (cast('2000-02-01' as date) +  interval '60' day)
 and i_manufact_id in (677,940,694,808)
 and inv_quantity_on_hand between 100 and 500
 and cs_item_sk = i_item_sk
 group by i_item_id,i_item_desc,i_current_price
 order by i_item_id
 limit 100;

DuckDB returns:
AAAAAAAAHFMBAAAA
Workers used to send inside long star methods. Profound, natural ministers wo.
86.11

According to the answer file provided by TPCDS, the correct answer is:
AAAAAAAAHFMBAAAA
Jobs crack at all public countries. Possible, traditional initiatives will make british problems. Fiscal years call extremely just growing pupils. Soon dutch voices
86.11

The second columns are different.

### OS:

Windows amd64

### DuckDB Version:

0.8.2 main

### DuckDB Client:

CLI

### Full Name:

Dan Cory

### Affiliation:

self

### Have you tried this on the latest `master` branch?

I have tested with a master build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by DanCory** [comment]
FYI go curious and this still repros in 1.3.1.

**Comment by DanCory** [comment]
The function DSDGen needs to call row_stop(table_id) at the end of every row. This calls checkseeds and may cycle each columns random number generators several times before the next row.

It appears this is part of dsdgen to ensure there are a known number of random numbers consumed per row. This enables starting and stopping a table at any row, which is used for threading.

Adding this call does slow down dsdgen but not noticeably.

**Comment by DanCory** [comment]
Fixed this issue, but found there were two other issues that had to be fixed before the extension would actually match the official dsdgen application:
1. The int overflow emulation in address.c is wrong. I just removed it. I'm not sure whether some version of it might be needed by some ports.
2. The code in tpcds_append_information::IsNull returns whether an output value should be null. It uses the column number in the appender, rather than the column number from columns.h. Unfortunately, these don't always match. The best fix I found was to pass in the column number to the append functions.
With all three fixes, the application exactly matches the application at sf=1. I haven't tested the others yet.
